bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Sensor.java

```diff
@@ -70,9 +72,14 @@ public Sensor(ComponentFactory.ComponentConfiguration componentConfiguration) {
 
         Value value;
         String uom = channelConfiguration.unitOfMeasurement;
+        String sc = channelConfiguration.stateClass;
 
         if (uom != null && !uom.isBlank()) {
             value = new NumberValue(null, null, null, UnitUtils.parseUnit(uom));
+        } else if (sc != null && !sc.isBlank()) {
```

Comment:
No need to check for "None"?

Reply:
No. `None` in the HASS docs isn't a literal string, but the Python `None`, which is equivalent to `null` in Java and JSON.

Reply:
Ok